pkg/cmd/pulumi/project/project_ls.go

```diff
@@ -0,0 +1,216 @@
+					org = getOrgFromStackName(stackFullName)
+
+					// Skip if we're filtering by organization and this stack doesn't match
+					if orgName != "" && org != orgName {
+						continue
+					}
+				}
+
+				// Update project entry
+				project, exists := projectMap[projectNameStr]
+				if !exists {
+					project = projectListResult{
+						Name:         projectNameStr,
+						Organization: org,
+						StackCount:   0,
+						Backend:      b.Name(),
+					}
+				}
+				project.StackCount++
+				projectMap[projectNameStr] = project
+			}
+
+			// Convert map to slice for output
+			var results []projectListResult
+			for _, project := range projectMap {
+				results = append(results, project)
+			}
+
+			// If no projects, display a message
+			if len(results) == 0 {
+				if orgName != "" {
+					fmt.Println("No projects found in organization", orgName)
+				} else {
+					fmt.Println("No projects found")
+				}
+				return nil
+			}
+
+			// Output the results
+			if jsonOut {
+				out, err := json.MarshalIndent(results, "", "    ")
+				if err != nil {
+					return err
+				}
+				fmt.Println(string(out))
+			} else {
+				// Display header based on backend type
+				if b.SupportsOrganizations() {
+					if orgName != "" {
+						fmt.Printf("PROJECTS IN ORGANIZATION %s:\n", orgName)
+					} else {
+						fmt.Println("PROJECTS:")
+					}
+
+					for _, result := range results {
+						if orgName == "" {
+							fmt.Printf("  %s (org: %s, stacks: %d)\n", result.Name, result.Organization, result.StackCount)
+						} else {
+							fmt.Printf("  %s (stacks: %d)\n", result.Name, result.StackCount)
+						}
+					}
+				} else {
+					// Display backend type for DIY backends
+					backendType := getDIYBackendType(b.URL())
+					fmt.Printf("PROJECTS (%s):\n", backendType)
+					for _, result := range results {
+						fmt.Printf("  %s (stacks: %d)\n", result.Name, result.StackCount)
+					}
+				}
+			}
+
+			return nil
+		},
+	}
+
+	cmd.PersistentFlags().StringVarP(
+		&orgName, "organization", "o", "", "The organization whose projects to list")
+	cmd.PersistentFlags().BoolVarP(
+		&jsonOut, "json", "j", false, "Emit output as JSON")
+
+	return cmd
+}
+
+// getOrgFromStackName extracts the organization name from a fully qualified stack name.
+// Stack names are typically in the format "org/project/stack" or just "project/stack".
+func getOrgFromStackName(stackName string) string {
+	parts := strings.Split(stackName, "/")
+	if len(parts) < 3 {
+		return "" // No organization in the stack name
+	}
+	return parts[0]
+}
+
+// getDIYBackendType returns a user-friendly name for the DIY backend type
+func getDIYBackendType(url string) string {
+	if strings.HasPrefix(url, "file://") {
+		return "local"
+	} else if strings.HasPrefix(url, "s3://") {
+		return "aws-s3"
+	} else if strings.HasPrefix(url, "azblob://") {
+		return "azure-blob"
+	} else if strings.HasPrefix(url, "gs://") {
+		return "google-cloud"
+	}
+	return "diy" // Generic fallback
```

Comment:
Not really seeing the value of this, `whoami -v` already tells you your backend url. I don't think the CLI needs to map that to a kind.